common/eth2/src/types.rs

```diff
@@ -1879,93 +1756,49 @@ impl<T: EthSpec, Payload: AbstractExecPayload<T>> TryFrom<SignedBeaconBlock<T, P
     }
 }
 
-impl<T: EthSpec, Payload: AbstractExecPayload<T>> From<SignedBlockContentsTuple<T, Payload>>
-    for SignedBlockContents<T, Payload>
-{
-    fn from(block_contents_tuple: SignedBlockContentsTuple<T, Payload>) -> Self {
+impl<T: EthSpec> From<SignedBlockContentsTuple<T>> for SignedBlockContents<T> {
+    fn from(block_contents_tuple: SignedBlockContentsTuple<T>) -> Self {
         SignedBlockContents::new(block_contents_tuple.0, block_contents_tuple.1)
     }
 }
 
 #[derive(Debug, Clone, Serialize, Deserialize, Encode)]
 #[serde(bound = "T: EthSpec")]
-pub struct SignedBeaconBlockAndBlobSidecars<T: EthSpec, Payload: AbstractExecPayload<T>> {
-    pub signed_block: SignedBeaconBlock<T, Payload>,
-    pub signed_blob_sidecars: SignedSidecarList<T, Payload::Sidecar>,
+pub struct SignedBeaconBlockAndBlobSidecars<T: EthSpec> {
```

Comment:
This struct no longer contain the list of blob sidecars, and we now have `SignedBlockContents` in the beacon API spec, but this name has already been taken by the enum wrapper. Is it worth renaming this (and the enum type)?

Reply:
That's a good point. 
I'm not sure `SignedBeaconBlockAndBlobs` would be apt either since it contains the proofs as well. 
We could say that the struct contains everything to be able to generate a `BlobSidecar` so the naming makes sense that way, but I'm open to renaming, just not sure to what.
cc @realbigsean

Reply:
Oh yea, naming this to match the api spec is smart. Hmm but then the wrapper name.. maybe `SignedBlockOrBlockContents` or `SignedBlockContentsWrapper`

Reply:
This commit contains some naming updates:
https://github.com/sigp/lighthouse/pull/4965/commits/9d7ac74702311cb42381f1be15c66b2cdea1c153

For now I went with this pattern, figuring `BlockContents` matching the spec is good. And naming the enum wrapping this with a `Full` prefix helps you understand where it should lie in the enum that wraps that (the blindedness enum).  I don't really like the generic-ness of the `Wrapper` suffix but couldn't think of a better blind vs. full type word:

```
pub enum BlockContentsWrapper<E: EthSpec> {
    Full(FullBlockContents<E>),
    Blinded(BlindedBeaconBlock<E>),
}

pub enum FullBlockContents<T: EthSpec> {
    BlockContents(BlockContents<T>),
    Block(BeaconBlock<T>),
}

pub struct BlockContents<T: EthSpec> {
    pub block: BeaconBlock<T>,
    pub kzg_proofs: KzgProofs<T>,
    pub blobs: BlobsList<T>,
}
```